src/VersionTool/Program.cs

```diff
@@ -225,13 +253,324 @@ private static int OnList(CommandOption pathOption)
                 {
                     SaveJObject(project.ProjectFilePath, root);
 
-                    Console.WriteLine($"Updated: {project.ProjectFilePath}");
+                    //Console.WriteLine($"Updated: {project.ProjectFilePath}");
                 }
             }
 
             return 0;
         }
 
+        private static int OnUpdatePatch(
+            CommandOption directoryOption,
+            CommandOption udpatePatchConfigOption,
+            CommandOption updatePackageOption,
+            CommandOption updateRepoOption,
+            CommandArgument patchConfigArgument)
+        {
+            var directory = directoryOption.Value();
+            var patchConfigPath = patchConfigArgument.Value;
+            var patchConfig = LoadJObject(patchConfigPath);
+
+            var repos = (patchConfig["Repos"] as JArray)?.Children()
+                .Select(token => new Repository((string)token, Path.Combine(directory, (string)token)))
+                .ToList();
+
+            // Handle patch config update
+            if (udpatePatchConfigOption.HasValue())
+            {
+                // Clear the Packages section
+                patchConfig["Packages"] = null;
+
+                // Scan for all packages and create new Packages section
+                UpdatePatchConfig(patchConfig, repos);
+                SaveJObject(patchConfigPath, patchConfig);
+
+                return 0;
+            }
+
+            var rules = (patchConfig["Rules"] as JArray)?.Children()
+                .Select(r => (r as JObject).ToObject<Rule>())
+                .ToList();
+            var packages = (patchConfig["Packages"] as JArray)?.Children()
+                .Select(r => (r as JObject).ToObject<Package>())
+                .ToList();
+
+            BuildRepoGraph(repos);
+
+            // Handle package updates
+            if (updatePackageOption.HasValue())
+            {
+                foreach (var packageUpdate in updatePackageOption.Values)
+                {
+                    var parsedPackageUpdate = packageUpdate.Split(new[] { ';' }, 2);
+
+                    if (parsedPackageUpdate.Length != 2)
+                    {
+                        throw new Exception($"Invalid package update option {packageUpdate}.");
+                    }
+
+                    var packageName = parsedPackageUpdate[0];
+                    var newPackageVersion = parsedPackageUpdate[1];
+
+                    var repo = repos.Where(r => r.Packages.Contains(packageName)).SingleOrDefault();
```

Comment:
Actually why `FirstOrDefault`? There shouldn't be duplicates and if there is, something has gone wrong. In that case I would prefer if the tool just barfs. On that note, I should do a bit more exception handling and argument validation.